M2/Macaulay2/packages/Triangulations.m2

```diff
@@ -0,0 +1,1616 @@
+    )
+
+-- The following is currently only for point sets
+-- and is probably slow for larger triangulations too.
+isTriangulation = method()
+isTriangulation(Matrix, List) := (M, tri) -> (
+    -- for the moment, we assume that M is a point set.
+    d := #(tri#0) - 1;
+    P := convexHull M;
+    M' := M || matrix{{numcols M: 1}};
+    volP := d! * volume P;
+    volP2 := sum for t in tri list abs det M'_t;
+    if volP != volP2 then (
+        << "volume is not correct: " << volP << " != " << volP2 << endl;
+        return false;
+        );
+    simplices := hashTable for t in tri list t => convexHull M_t;
+    for x in subsets(keys simplices, 2) do 
+        if dim(intersection(simplices#(x#0), simplices#(x#1))) == d then (
+            << "simplices " << x << " intersect in full dimensional region" << endl;
+            return false;
+            );
+    true
+    )
+
+isRegularTriangulation = method(Options => {Homogenize => true})
+isRegularTriangulation Triangulation := Boolean => opts -> T -> (
+    topcomIsRegularTriangulation(T.cache.matrix, max T, Homogenize => false)
+    )
+isRegularTriangulation(Matrix, List) := Boolean => opts -> (A, tri) -> (
+    topcomIsRegularTriangulation(A, tri, opts)
+    )
+
+regularTriangulationWeights = method(Options => options isRegularTriangulation)
+regularTriangulationWeights Triangulation := List => opts -> T -> (
+    topcomRegularTriangulationWeights(matrix T, max T, Homogenize => false)
+    )
+regularTriangulationWeights(Matrix, List) := List => opts -> (A, tri) -> (
+    topcomRegularTriangulationWeights(A, tri, opts)
+    )
+
+regularFineTriangulation = method(Options => options isRegularTriangulation)
+regularFineTriangulation Matrix := Triangulation => opts -> (A) -> (
+    tri := topcomRegularFineTriangulation(A, opts);
+    if tri === null then null else 
+        triangulation(A, tri, opts)
+    )
+
+-- TODO/BUG: this ASSUMES (A, tri) is a triangulation.
+-- TODO: add in Homogenize as an option?
+isFine = method()
+isFine(Matrix, List) := Boolean => (A, tri) -> (
+    numcols A == tri//flatten//unique//length
+    )
+
+-- TODO/BUG: this ASSUMES (A, tri) is a triangulation.
+isStar = method()
+isStar(Matrix, List) := Boolean => (A, tri) -> (
+    -- assumption?  last column of A is the zero element?  (or the one the star is taken with respect to).
+    origin := numcols A - 1;
+    all(tri, t -> member(origin, t))
+    )
+
+isStar Triangulation := Boolean => T -> isStar(matrix T, max T)
+isFine Triangulation := Boolean => T -> isFine(matrix T, max T)
+
+allTriangulations = method(Options => options topcomAllTriangulations)
+allTriangulations Matrix := List => opts -> A -> (
+    tris := topcomAllTriangulations(A, opts);
+    for t in tris list triangulation(A, t, Homogenize => opts.Homogenize)
+    )
+
+-- TODO: this routine should be submitted to Polyhedra as a bug fix.
+regularSubdivision (Matrix,Matrix) := (M,w) -> (
```

Comment:
@mahrud This function ```regularSubdivision``` has now been fixed in Polyhedra by @lkastner, and so we should remove this one.  Perhaps it is easiest if you remove this in your repo?